.github/workflows/main.yml

```diff
@@ -62,9 +62,44 @@ jobs:
           submodules: recursive
 
       - name: Go test
-        run: go test -coverprofile coverage.out -timeout 28m ./...
+        run: go test -shuffle=on -coverprofile coverage.out -timeout 28m ./...
 
       - name: Upload coverage file to Codecov
         uses: codecov/codecov-action@v3
         with:
           files: coverage.out
+
+  mutating:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.18.x
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Install dependencies
+        run: make install
+
+      - name: Mutating testing
+        run: make mut
+
+#  test-race:
```

Comment:
Why not use the `-race` flag in the existing workflow? https://github.com/0xPolygon/go-ibft/pull/28/files#diff-7829468e86c1cc5d5133195b5cb48e1ff6c75e3e9203777f6b2e379d9e4882b3R65